DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift

```diff
@@ -397,17 +456,28 @@ extension DuckPlayerNavigationHandler: DuckNavigationHandling {
     @MainActor
     func handleAttach(webView: WKWebView) {
         
+        Logger.duckPlayer.debug("DP: Attach WebView")
+        
         guard featureFlagger.isFeatureOn(.duckPlayer) else {
             return
         }
         
         if let url = webView.url, url.isDuckPlayer,
             !url.isDuckURLScheme,
-            duckPlayer.settings.mode == .enabled || duckPlayer.settings.mode == .alwaysAsk {
+            duckPlayerMode == .enabled || duckPlayerMode == .alwaysAsk {
             Logger.duckPlayer.debug("DP: Handling Initial Load of a video for \(url.absoluteString)")
             handleReload(webView: webView)
         }
         
     }
-
+    
+    // Handle custom events
+    func handleEvent(event: DuckPlayerNavigationEvent, url: URL?, navigationAction: WKNavigationAction?) {
```

Comment:
The idea here is to handle 'Custom' events that are unrelated to DuckPlayer as a feature but not with presentation.-